lua/neo-tree/utils.lua

```diff
@@ -349,13 +349,17 @@ end
 ---@param sourceObject table The table to set a value in.
 ---@param valuePath string The path to the value to set.
 ---@param value any The value to set.
-M.set_value = function(sourceObject, valuePath, value)
+---@param force_tabulize boolean? Overwrite existing non-table parents with a table along the way.
```

Comment:
I don't think that the `force_tabulize` needs to exist because there is no reason for it to ever be false.

Reply:
I thought someone might want to call it with `pcall` with `force_tabularize = false` to bail out if type is different, and so that it doesn't overwrite the old value.

Reply:
Following that logic, I would make it `allow_type_change` to cover all cases where setting a value may change a type and not just specifically handle the table case only. I don't see this being needed at all though so I would remove the option because it just adds unnecessary complication.

Reply:
Got it! I'll just remove it and make it the default behavior :)